src/eventcontroller/EventController.ts

```diff
@@ -20,7 +38,13 @@ export default interface EventController {
   ): Promise<void>;
 
   /**
-   * Adds a state to the meeting history stack.
+   * A function that can be called to push a meeting state to the meeting history i.e. push a meeting event to the event
+   * history. This can be used to update the history without notifying EventObservers. It will also report to the Amazon Chime backend without notifying event observers.
+   */
+  pushMeetingState?(state: MeetingHistoryState): Promise<void>;
```

Comment:
Why is this optional?

Reply:
Not really core to the interface, currently we don't use it so if a builder doesn't wanna implmen

Reply:
> Not really core to the interface, currently we don't use it so if a builder doesn't want to implement it, there's no loss of functionality.

Reply:
Should we just remove it then if it is not used? @devalevenkatesh

Reply:
I think based on ingestion service request we changed or updated events to use `publishEvent` over `pushMeetingState`. I see earlier there were reference to this:
- https://github.com/aws/amazon-chime-sdk-js/blob/v2.12.0/src/audiovideocontroller/DefaultAudioVideoController.ts#L998
- https://github.com/aws/amazon-chime-sdk-js/blob/v2.12.0/src/task/MonitorTask.ts#L348

Would it be good to keep it now and audit later if we still need it?

Reply:
If it is not used. It is deadcode. We can always bring it back later if needed :D

Reply:
Ok to remove and agree we can add it back if required.

Reply:
Will remove then

Reply:
Yes, lets remove, found the PR below which added all events to `publishEvent` from `pushMeetingState`:
https://github.com/aws/amazon-chime-sdk-js/pull/1580